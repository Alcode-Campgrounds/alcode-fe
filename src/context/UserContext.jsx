import { useContext, useState, createContext, useEffect } from 'react';
import { getCurrentUser } from '../utils/AuthFetch';

const UserContext = createContext();

const UserProvider = ({ children }) => {
    const [user, setUser] = useState(null)
    const [loading, setLoading] = useState(true)

    useEffect(() => {
        getCurrentUser()
            .then(user => setUser(user))
            .catch(error => console.log(error))
            .finally(() => setLoading(false))
    }, [])

    return (
        <UserContext.Provider value={{ user, setUser, loading }}>
            {children}
        </UserContext.Provider>
    )
}

const useLoadingUser = () => {
    const { loading } = useContext(UserContext)
    return loading
}

const useUser = () => {
    const context = useContext(UserContext);
    if (context === undefined) {
        throw new Error('useUser hook must be called inside of a GuessContext Provider');
    }
    return context;
}

//placed here for comparison
const useAuth = () => {
    // useUser is its own custom hook and not a context. 
    // This useAuth isn't doing anything useUser isn't. It's returning the same context
    const context = useContext(useUser)

    if (context === undefined) {
        throw new Error('useAuth must be used within a provideAuth context');
    }

    return context;
}

export { UserProvider, useUser, useLoadingUser }
