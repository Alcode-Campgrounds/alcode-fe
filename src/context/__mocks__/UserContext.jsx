import { useContext, useState, createContext, useEffect } from 'react';
// import { getCurrentUser } from '../utils/AuthFetch';

const UserContext = createContext();

const UserProvider = ({children}) => {
    const [user, setUser] = useState(null)
    const [loading, setLoading] = useState(true)
    
    useEffect(() => { 
        setUser({name: 'fake-user', email: 'fake@email.com'})
        setLoading(false)
            // getCurrentUser()
            //     .then(user => setUser(user))
            //     .catch(error => console.log(error))
            //     .finally(()=>setLoading(false)) 
    }, [])

    return (
        <UserContext.Provider value={{user, setUser, loading}}>
            {children}
        </UserContext.Provider>
    )
}
const useLoadingUser = () => {
    const { loading } = useContext(UserContext)
    return loading
}
const useUser = () => { 
    const context = useContext(UserContext);
    if (context === undefined){
        throw new Error('useUser hook must be called inside of a GuessContext Provider');
    }
    return context;
}

export { UserProvider, useUser, useLoadingUser}