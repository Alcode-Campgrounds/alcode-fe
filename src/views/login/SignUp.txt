// import React from 'react'
// import { create }from 'react-test-renderer'
// import { UserProvider } from '../../context/UserContext'
// import SignUp from './SignUp'
// import { MemoryRouter as Router, Routes, Route } from 'react-router-dom'
 

 



// describe('is it here?', () => {
//     it('did it render?', () => {
    
//     const wrapper = create(
//     <UserProvider>
//         <Router initialEntries={["/signup"]}>
//             <Routes>
//                 <Route path="signup" element={<SignUp />} />
//             </Routes>
//         </Router>
//     </UserProvider>
//     )

//         expect(wrapper).toMatchSnapshot()
//     })
// })

