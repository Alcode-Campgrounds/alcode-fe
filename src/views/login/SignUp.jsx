import React from 'react';
import { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { useUser } from '../../context/UserContext';
import { existingUserSignIn, newUserSignUp } from '../../utils/AuthFetch';

export default function SignUp({ hasUser = false }) {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [name, setName] = useState('');
  const [error, setError] = useState('');
  // const [gooBtn, setGooBtn] = useState(false)
  const { setUser } = useUser();

  let navigate = useNavigate();

  const handleSubmitSignUp = async (e) => {
    e.preventDefault();

    if (hasUser) {
      const userData = await existingUserSignIn(email, password);
      if (userData.email) {
        setUser({ name: userData.name, email: userData.email });
        navigate('/search');
      } else {
        setError('Wrong email/password');
      }
    } else {
      try {
        const userSignUp = await newUserSignUp(name, email, password);
        console.log(userSignUp.status);
        if (userSignUp) {
          setUser({ name, email });
          navigate('/search');
        }
      } catch (error) {
        setError(error.message);
      }
    }
  };

  return (
    <form className='form-main' onSubmit={handleSubmitSignUp}>
      <fieldset>
        <legend className='legend'>{hasUser ? 'SignIn' : 'SignUp'}</legend>
      
          <a className='google-btn' href={process.env.REACT_APP_OAUTH}>
          Google
          </a>
        <label htmlFor='email-input'>
          <input
            id='email-input'
            required
            value={email}
            type='text'
            placeholder='email'
            className='input-A'
            onChange={({ target }) => {
              setEmail(target.value);
            }}
          />
        </label>
        <label htmlFor='password-input'>
          <input
            id='password-input'
            required
            value={password}
            type='text'
            placeholder='password'
            className='input-B'
            onChange={({ target }) => {
              setPassword(target.value);
            }}
          />
        </label>

        {!hasUser && (
          <label htmlFor='name-input'>
            <input
              id='name-input'
              required
              value={name}
              type='text'
              placeholder='name'
              className='input-C'
              onChange={({ target }) => {
                setName(target.value);
              }}
            />
          </label>
        )}
        <br />
        <button type='submit' className='primary-btn'>
          {hasUser ? 'SignIn' : 'SignUp'}
        </button>
        <p>{error}</p>
      </fieldset>
    </form>
  );
}
