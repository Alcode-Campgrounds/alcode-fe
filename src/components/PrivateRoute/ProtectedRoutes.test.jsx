import * as React from "react";
import { create } from  'react-test-renderer'
import {
  MemoryRouter,
  Routes,
  Route
} from "react-router-dom";
import { UserProvider } from '../../context/UserContext';
import SignUp from '../../views/login/SignUp';

describe("private route", () => {
  it("renders sign in route", () => {
 let renderer = create(
     <UserProvider>
      <MemoryRouter initialEntries={["/signin"]}>
        <Routes>
            <Route path="signin" element={<SignUp hasUser />}/>
         
        </Routes>
      </MemoryRouter>
      </UserProvider>
    );

    expect(renderer.toJSON()).toMatchSnapshot();

  });
});

// NEED TO GUIDE BROWSER THROUGH THE SIGN IN PROCESS TO RENDER A SEARCH PAGE 