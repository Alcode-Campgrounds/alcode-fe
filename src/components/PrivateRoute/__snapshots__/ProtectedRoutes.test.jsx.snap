// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`private route renders sign in route 1`] = `
<fieldset>
  <legend>
    SignIn
  </legend>
  <br />
  <form
    className="form-main"
    onSubmit={[Function]}
  >
    <a
      href="https://nameless-taiga-88760.herokuapp.com/api/alcode/auth/login"
    >
      Sign in with your Google account
    </a>
    <label
      htmlFor="email-input"
    >
      <input
        className="input"
        id="email-input"
        onChange={[Function]}
        placeholder="email"
        required={true}
        type="text"
        value=""
      />
    </label>
    <label
      htmlFor="password-input"
    >
      <input
        className="input"
        id="password-input"
        onChange={[Function]}
        placeholder="password"
        required={true}
        type="text"
        value=""
      />
    </label>
    <br />
    <button
      className="btn-main"
      type="submit"
    >
      SignIn
    </button>
    <br />
    <p>
      
    </p>
  </form>
</fieldset>
`;
